### PR TITLE
upgrade github actions and add dependabot config file

### PR DESCRIPTION
**Pull request by AdrianDsg**
Hi,
I just noticed that several of the GitHub Actions used by this project are apparently outdated.
I also added a dependabot config file to ease the maintainance of the dependencies.
Regards,
Adrian

### CODE REVIEW

**Comment by shawnHartsell** [comment]
@AdrianDsg thank you for the contribution! This repo has changed hands internally which has caused this PR to fall through the cracks. We're currently working on housekeeping items to get this project back into shape (updating Go, deps, etc). After that's completed we should be able to circle back around to this PR.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: 2
+updates:
+  # Maintain dependencies for GitHub Actions
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: daily
```
Comment (joshmgross) [inline .github/dependabot.yml:7]: Daily updates could get noisy, maybe we should do weekly?

File: .github/workflows/releases.yml [hunk 2]
```diff
@@ -10,15 +10,15 @@ jobs:
...
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
       - name: Set up Go
         uses: actions/setup-go@v4
       - name: Set CURRENT_TAG
         run: echo "GORELEASER_CURRENT_TAG=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
       - name: Run GoReleaser
-        uses: goreleaser/goreleaser-action@v3
+        uses: goreleaser/goreleaser-action@v5
```
Comment (joshmgross) [inline .github/workflows/releases.yml:21]: We should pin 3rd party actions to a full SHA - https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

**Comment by shawnHartsell** [comment]
I'm going to go ahead and approve and merge even though the license check is failing. I noticed similar issues when merging in dependabot updates in the latest release.

**Pull request merged**